compiler-tests/src/test/kotlin/dev/zacsweers/metro/compiler/AbstractIrDumpTest.kt

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2025 Zac Sweers
+// SPDX-License-Identifier: Apache-2.0
+package dev.zacsweers.metro.compiler
+
+import org.jetbrains.kotlin.config.JvmTarget
+import org.jetbrains.kotlin.test.backend.handlers.IrPrettyKotlinDumpHandler
+import org.jetbrains.kotlin.test.backend.handlers.IrTextDumpHandler
+import org.jetbrains.kotlin.test.builders.TestConfigurationBuilder
+import org.jetbrains.kotlin.test.builders.configureIrHandlersStep
+import org.jetbrains.kotlin.test.directives.CodegenTestDirectives.DUMP_IR
+import org.jetbrains.kotlin.test.directives.CodegenTestDirectives.DUMP_KT_IR
+import org.jetbrains.kotlin.test.directives.CodegenTestDirectives.IGNORE_DEXING
+import org.jetbrains.kotlin.test.directives.ConfigurationDirectives.WITH_STDLIB
+import org.jetbrains.kotlin.test.directives.JvmEnvironmentConfigurationDirectives.FULL_JDK
+import org.jetbrains.kotlin.test.directives.JvmEnvironmentConfigurationDirectives.JVM_TARGET
+import org.jetbrains.kotlin.test.runners.ir.AbstractFirLightTreeJvmIrTextTest
+import org.jetbrains.kotlin.test.services.KotlinStandardLibrariesPathProvider
+
+open class AbstractIrDumpTest : AbstractFirLightTreeJvmIrTextTest() {
+  override fun createKotlinStandardLibrariesPathProvider(): KotlinStandardLibrariesPathProvider {
+    return ClasspathBasedStandardLibrariesPathProvider
+  }
+
+  override fun configure(builder: TestConfigurationBuilder) {
+    super.configure(builder)
+
+    with(builder) {
+      configurePlugin()
+
+      defaultDirectives {
+        JVM_TARGET.with(JvmTarget.JVM_11)
+        +FULL_JDK
+        +WITH_STDLIB
+
+        +IGNORE_DEXING // Avoids loading R8 from the classpath.
+
+        -DUMP_IR
+        +DUMP_KT_IR
```

Comment:
I disabled the default `irElement.dump()` directive (`DUMP_IR`) and enabled the non-default `irElement.dumpKotlinLike()` directive instead (`DUMP_KT_IR`). You can explicitly enable the `DUMP_IR` directive in a test though with `// DUMP_IR` at the top of the file if you want to see the more detailed IR dump.